### ISSUE TITLE
Suggestions for finetuning the model on custom dataset

### ISSUE THREAD

**Issue by Dylan-H-Wang** [Mobile-Sensing-and-UbiComp-Laboratory/NormWear#5]
Hi,

Thank you for the awesome work! I am currently running this repo for my custom PPG dataset (well not mine but public dataset, you can find here [Website](https://sites.google.com/view/ybenezeth/ubfcrppg) and [Google Drive](https://drive.google.com/drive/folders/1o0XU4gTIo46YfwaWjIgbtCncc-oF44Xk))

I am running the `downstream_pipeline.main_finetune.py` using `normwear_last_checkpoint-15470-correct.pth` with `lr=1e-3`, `batch_size=16`, `max_len=390`. Other settings are keep default.

The data is preprocessed into length `180` with sampling rate `30`, which means the mini-batch shape is `16x180` during the training. These processed data were fed into the NormWear and was asked to output corresponding heart rate values with shape `16x1` (by setting `num_classes=1`).

The task is regression which predicts the heart rate for this PPG signal segment. The ground truth heart rate label is calculated based on Fast Fourier transform (FFT) method.

However, the mean absolute error after training is around 5. Any suggestion for the finetune process?

Thank you for the help and time!

**Comment by yunfeiluo** [comment]
Hi,

Thank you for following up on our work!

As a few suggestions, we would first recommend trying inference without fine-tuning to establish a performance baseline. For example, by using different types of regressors on top of the frozen model. Fine-tuning the entire model can easily lead to overfitting, depending on the characteristics of your dataset. 

Also, an MAE of ~5 bpm may not be unreasonable, depending on the signal quality and the specific setup of your task. Please note that NormWear was originally pretrained on signals with a 65 Hz sampling rate, so using a consistent rate might help improve the quality of the initial representations. 

Hope this helps, and feel free to share more details if you continue experimenting!

Best regards,
NormWear Team
